cg/store/api/findbasicdata.py

```diff
@@ -68,6 +68,10 @@ def customer_by_id(self, id_: int) -> models.Customer:
         """Fetch a customer by id number from the store."""
         return self.Customer.query.filter_by(id=id_).first()
 
+    def sample_by_id(self, id_: int) -> models.Sample:
```

Comment:
Would it be possible to reuse:
```
   def sample(self, internal_id: str) -> models.Sample:
        """Fetch a sample by lims id."""
        return self.Sample.query.filter_by(internal_id=internal_id).first()
```
Could you refactor this filter into `cg/store/status_sample_filters.py` and decouple the query into a separate function much like:
```
def _get_case_query(self) -> Query:
        """Return case query."""
        return self.query(Family)
```
This will facilitate keeping it DRY and facilitate unit testing and changing the models in the future

Reply:
Something like this?

```
    def _get_sample_query(self) -> Query:
        """Return sample query."""
        return self.query(Sample)

    def sample(self, internal_id: str) -> Sample:
        """Fetch a sample by lims id."""
        sample: Sample = apply_sample_filter(
            "get_sample_by_internal_id", samples=self._get_sample_query(), sample_id=internal_id
        )
```

Reply:
I'm not sure how nice it is to have these filter name strings. Would it not be better to at least have them as an enum to make their names easier to change and use?

Reply:
That could be a good idea 👍

Reply:
I think this PR might become too extensive refactoring it now, we'll do it another. But I extracted the filter functions and added tests for them.